vt2pbf/service/feature.py

```diff
@@ -41,17 +41,35 @@ def add_tags(self, tags: dict):
             if v is None:
                 continue
 
+            # False and True are equal to 0 and 1, respectively, when used as
+            # keys to Python `dict`s, see https://stackoverflow.com/q/22275027
+            #
+            # To avoid collision, store the value as a tuple of type and value
+            # This applies to boolean values, only, and only affects the look-up
+            # table in self._layer.value_indices, not the values written to
+            # protobuf
+
+            # To align the values with the output of the JavaScript
+            # implementation, in which full-number float values and int values
+            # are deduplicated, let’s also cast int to float
+            if isinstance(v, bool):
+                v_ = (type(v), v)
+            elif isinstance(v, int):
```

Comment:
I wrote this with if/elif/else instead of the suggested ternaries, as `isinstance(v, int) == True` for boolean `v`